src/tests/integration_test_utils.rs

```diff
@@ -119,23 +126,47 @@ pub fn lines_match(expected: &str, have: &str, skip: Option<usize>) {
         assert_eq!(
             &expected[indentation..],
             it.next().unwrap(),
-            "on line {} of input",
-            i + 1
+            "on line {} of input:\n{}",
+            i + 1,
+            delineated_string(have),
         );
     }
-    assert_eq!(it.next(), None, "more input than expected");
+    if !allow_partial {
+        assert_eq!(it.next(), None, "more input than expected");
+    }
+}
+
+pub fn lines_match(expected: &str, have: &str) {
+    lines_match_skip(0, expected, have)
```

Comment:
Can we call these `assert_lines_match` and `assert_lines_match_after_skip`? Sorry, you appear to have opened a Pandora's box... :)

Reply:
These are nice and clear.  I had contemplated changing expect_skip() to expect_after(), but I was already worried about going overboard on tweaking someone else's code. 😄 Your suggestion of skip -> after_skip is even clearer, as is the assert prefix for the lines_match functions.  I'm also changing expect_skip() to expect_after_skip().